### PR TITLE
translate IntSum to Sum in otlp_wrappers

### PR DESCRIPTION
**Pull request by codeboten**
**Description:**
Deprecating IntSum As per https://github.com/open-telemetry/opentelemetry-proto/blob/f3b0ee0861d304f8f3126686ba9b01c106069cb0/opentelemetry/proto/metrics/v1/metrics.proto#L156, the OTLP receiver will translate `IntSum` into `Sum`. 

**Link to tracking Issue:** Part of #3534 

**Testing:** Added a test to ensure both `Sum` and `IntSum` are valid `Sum` after translation.

This change is built on top of #3619, will rebase once merged.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Tests failing:

```
=== RUN   TestSendMetrics
    otlp_test.go:295: 
        	Error Trace:	otlp_test.go:295
        	Error:      	Not equal: 
        	            	expected: pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc00000e380)}
        	            	actual  : pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc0003aa860)}
        	            	
        	            	Diff:
        	Test:       	TestSendMetrics
```

**Comment by bogdandrutu** [comment]
I think that is a good sign, it seems that the otlp receiver executes the code :)

File: exporter/otlpexporter/otlp_test.go [hunk 1]
```diff
@@ -277,7 +277,7 @@ func TestSendMetrics(t *testing.T) {
 	assert.EqualValues(t, 0, atomic.LoadInt32(&rcv.totalItems))
 
 	// A trace with 2 spans.
```
Comment (bogdandrutu) [inline exporter/otlpexporter/otlp_test.go:279]: Update comment?

File: exporter/otlpexporter/otlp_test.go [hunk 2]
```diff
@@ -277,7 +277,7 @@ func TestSendMetrics(t *testing.T) {
...
-	md = testdata.GenerateMetricsTwoMetrics()
+	md = testdata.GenerateMetricsTwoSumMetrics()
```
Comment (bogdandrutu) [inline exporter/otlpexporter/otlp_test.go:280]: Instead of adding new method maybe consider to change the other one and add same TODO.

Comment (codeboten) [reply exporter/otlpexporter/otlp_test.go:280]: I thought adding another func would be less invasive, but it turns out I was wrong. Will add the same TODO to all the places I change here.

File: exporter/otlpexporter/otlp_test.go (hunk 1 above)
Comment (codeboten) [reply exporter/otlpexporter/otlp_test.go]: Updated!

File: exporter/otlpexporter/otlp_test.go (hunk 2 above)
Comment (codeboten) [reply exporter/otlpexporter/otlp_test.go:280]: @bogdandrutu I ended up adding 2 temporary test methods, changing the existing one caused all sorts of problems with other tests that were depending on them.

I plan on tackling the IntDataPoint/DoubleDataPoint conversion to NumberDataPoint soon, which will allow us to remove these temporary functions.

**Comment by bogdandrutu** [comment]
Please rebase

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**